neuralprophet/forecaster.py

```diff
@@ -3288,12 +3296,30 @@ def _reshape_raw_predictions_to_forecst_df(self, df, predicted, components, pred
                     forecast_rest = components[comp][1:, self.n_forecasts - 1, j]
                     yhat = np.concatenate(([np.NaN] * self.max_lags, forecast_0, forecast_rest))
                     if prediction_frequency is not None:
-                        forecast_rest = components[comp][1:, :, j]
-                        yhat = np.concatenate(([np.NaN] * self.max_lags, forecast_0, forecast_rest.flatten()))
-                        if len(yhat) < len(df_forecast):
-                            yhat = np.concatenate((yhat, [np.NaN] * (len(df_forecast) - len(yhat))))
-                        else:
-                            yhat = yhat[: len(df_forecast)]
+                        for key, value in prediction_frequency.items():
+                            if key == "daily":
```

Comment:
To make the unit to be provided clear, I suggest to add it to the key, e.g. "daily-hour"